db/view/view.cc

```diff
@@ -1714,14 +1723,17 @@ class view_builder::consumer {
     }
 
     stop_iteration consume(tombstone) {
+        inject_failure("view_builder_consume_tombstone");
         return stop_iteration::no;
     }
 
     stop_iteration consume(static_row&&, tombstone, bool) {
+        inject_failure("view_builder_consume_static_row");
```

Comment:
I have two questions about injection for such very deep functions:
1. Does it not hurt performance, given that injections points are recognized by strings?
2. You would have millions of calls to this consume(). Will it fail probabilistically? Or just the first call will fail? Or what?

Reply:
1. Release builds have error injection statically disabled - which effectively makes all calls to `inject_failure` empty functions, with the assumption that they should be easily recognizable by the compiler as such, and removed altogether.
2. It's literally whatever you want - an injection point can be turned on and off via REST API calls, and they can be either one-shot or always-on. In the first dtest I prepared I used the one-shot variant, but there can also be scenarios when you want the call to perpetually fail for a time, e.g. to make sure that view building will continue after the call eventually succeeds, even if it takes a couple of seconds/minutes.